pkg/asset/manifests/dns.go

```diff
@@ -54,7 +54,7 @@ func (d *DNS) Generate(dependencies asset.Parents) error {
 			// not namespaced
 		},
 		Spec: configv1.DNSSpec{
-			BaseDomain: installConfig.Config.BaseDomain,
+			BaseDomain: installConfig.Config.ClusterDomain(),
```

Comment:
Do we instead need to change `DNSSpec` like:

```go
type DNSSpec struct {
  BaseDomain string `json:"baseDomain"`
  ClusterDomain string `json:"baseDomain"`
}
```

Otherwise downstream consumers (e.g. the ingress operator's DNS manager) are left to guess the public zone.

Reply:
> Do we instead need to change `DNSSpec` like:
> 
> ```go
> type DNSSpec struct {
>   BaseDomain string `json:"baseDomain"`
>   ClusterDomain string `json:"clusterDomain"`
> }
> ```

I think the `Basedomain` field correctly represents what is required.
from [doc](https://github.com/openshift/api/blob/d67473e7f1907b74d1f27706260eecf0bc9f2a52/config/v1/types_dns.go#L23-L27)
```
	// baseDomain is the base domain of the cluster. All managed DNS records will
	// be sub-domains of this base.
	//
	// For example, given the base domain `openshift.example.com`, an API server
	// DNS record may be created for `cluster-api.openshift.example.com`.
```

while the Ingress spec only specifies the `domain` used for routes
from [doc](https://github.com/openshift/api/blob/d67473e7f1907b74d1f27706260eecf0bc9f2a52/config/v1/types_ingress.go#L23-L26)
```
	// domain is used to generate a default host name for a route when the
	// route's host name is empty.  The generated host name will follow this
	// pattern: "<route-name>.<route-namespace>.<domain>".
```
but does not specify how/where to realize that information for ingress's DNS manager.

Reply:
So what if we extend the `DNSSpec` to 
```go
type DNSSpec struct {
  BaseDomain string `json:"baseDomain"`
  PublicZone *DNSZone `json:"publicZone"`
  PrivateZone *DNSZone `json:"privateZone"`
}

// DNSZone describes a dns zone for a provider.
// A zone can be identified by an identifier or tags.
type DNSZone struct {
  // id is the identifier that can be used to find the dns zone
  // +optional
  ID *string `json:"id"`
  // tags is a map of tags that can be used to query the dns zone
  // +optional
  Tags map[string]string `json:"tags"`
}
```

Reply:
@ironcladlou https://github.com/openshift/api/pull/202

Reply:
https://github.com/openshift/cluster-ingress-operator/pull/121 covers step 3.